### DISCUSSION TITLE
Need a working UI for prez v4

### DISCUSSION THREAD

**Discussion by luoenzhen** [RDFLib/prez#306] [Q&A]
1. Follow https://github.com/RDFLib/prez/issues/303, I need a working prez UI for prez v4.

2. Does the file `prez/reference_data/profiles/vocprez_default_profiles.ttl` from v3 become v4's `prez/reference_data/profiles/prez_default_profiles.ttl`?

**Comment by recalcitrantsupplant** [comment]
Can you check back with us in a week re. 1. The short answer is it works but we need to document it better.

Re. 2, no, the Profiles defined for v3 -> v4 have changed. V4 tries to reuse SHACL predicates as much as possible. It is similar to what you have in V4. In v4 on the relevant profile you will add all the predicates you want to include under a sh:union like this:

```turtle
    sh:property [
        sh:path (
            sh:union (
                  rdf:type
                  dcterms:publisher
                  reg:status
                  ( prov:qualifiedDerivation prov:hadRole )
                  ( prov:qualifiedDerivation prov:entity )
            )
        )
    ] ;
```

The full documentation is here https://github.com/RDFLib/prez/blob/main/README-Dev.md#combinations-of-paths

**Reply by luoenzhen** [reply]
Thank you for the quick response. I will check back with you in a week.

**Reply by luoenzhen** [reply]
Appendix D is not found: https://github.com/RDFLib/prez/blob/main/appendix-d---example-profile-and-mediatype-selection-sparql-query.

**Reply by recalcitrantsupplant** [reply]
what's the specific info you're after? some of the links need updating

**Reply by luoenzhen** [reply]
I want to know the sparql query to get the profile from triple store.

**Reply by recalcitrantsupplant** [reply]
Prez doesn't write to the triplestore at all; only data you add for profiles will be there

**Comment by recalcitrantsupplant** [comment]
Copying Howard's response here:

> Hi @luoenzhen. We are still in the process of documenting and finalising the UI app for v4. However, in the meantime you can point our v4 demo UI instance to any Prez API endpoint to test how it renders.
>
> Steps:
>
> Go to the endpoint test page --> https://demo.dev.kurrawong.ai/endpoint-tester
Enter your Prez API endpoint
Press "Set" (this should show that you are using a custom endpoint now)
Go to catalogs --> https://demo.dev.kurrawong.ai/catalogs
You should see catalogs from your Prez API instance
If you use a http not https Prez API endpoint you may hit a browser security issue. If that's a problem, maybe we can make a http version accessible as well.
>
> Let me know how it goes.

To use your data with the v4 API without making any configuration changes, you will need to add some additional triples to your data.
You will need to add a dcat:Catalog which dcterms:hasPart each of your skos:ConceptScheme and skos:Collection.
Once you have done that, the UI Howard linked to should work with the v4 API.
